frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx

```diff
@@ -0,0 +1,143 @@
+  MenuList,
+  TextContent,
+  TextVariants,
+} from '@patternfly/react-core';
+import React from 'react';
+import { EyeIcon, KeyIcon } from '@patternfly/react-icons';
+import { DataConnection, AWSDataEntry } from '~/pages/projects/types';
+import {
+  convertAWSSecretData,
+  getDataConnectionDisplayName,
+} from '~/pages/projects/screens/detail/data-connections/utils';
+import { PIPELINE_AWS_KEY } from '~/pages/projects/dataConnections/const';
+import { PipelineServerConfigType } from './types';
+import { getLabelName } from './utils';
+type pipelineDropdownProps = {
+  setConfig: (config: PipelineServerConfigType) => void;
+  config: PipelineServerConfigType;
+  dataConnections: DataConnection[];
+};
+export const PipelineDropdown = ({ config, setConfig, dataConnections }: pipelineDropdownProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [passwordHiddenStates, setPasswordHiddenStates] = React.useState<{
+    [key: number]: boolean;
+  }>(
+    dataConnections.reduce((acc: { [key: number]: boolean }, _, index) => {
+      acc[index] = true;
+      return acc;
+    }, {}),
+  );
+  const existingDataConnection = (connection: string): AWSDataEntry | null => {
+    const value = dataConnections.find((d) => d.data.metadata.name === connection);
+
+    if (value) {
+      return convertAWSSecretData(value).filter((dataItem) =>
+        PIPELINE_AWS_KEY.some((filterItem) => filterItem === dataItem.key),
+      );
+    }
+    return null;
+  };
+  const onToggle = (isOpen: boolean) => {
+    setPasswordHiddenStates(
+      dataConnections.reduce((acc: { [key: number]: boolean }, _, index) => {
+        acc[index] = true;
+        return acc;
+      }, {}),
+    );
+    setIsOpen(isOpen);
+  };
+
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    option?: string | number | null,
+  ) => {
+    if (typeof option === 'string') {
+      const optionValue = existingDataConnection(option);
+      const updatedObjectStorageValue = config.objectStorage.newValue.map((item) => {
+        const matchingOption = optionValue?.find((optItem) => optItem.key === item.key);
+
+        return {
+          ...item,
+          value: matchingOption ? matchingOption.value : item.value,
+        };
+      });
+
+      setConfig({
+        ...config,
+        objectStorage: {
+          newValue: updatedObjectStorageValue,
+        },
+      });
+    }
+  };
+  return (
+    <Dropdown
+      menuAppendTo="parent"
+      position={DropdownPosition.right}
+      toggle={<DropdownToggle onToggle={onToggle} icon={<KeyIcon />}></DropdownToggle>}
+      isOpen={isOpen}
+    >
+      <Menu onSelect={onSelect} isScrollable isPlain>
+        <MenuContent>
+          <MenuGroup
+            labelHeadingLevel="h3"
+            label={
+              <Text component={TextVariants.p}>
+                <KeyIcon /> Populate the form with credentials from your selected data connection
+              </Text>
+            }
```

Comment:
Seems there's some bad code in the `MenuGroup` when it comes to supplying custom titles. 

I think this would mimic the default styling:

```suggestion
import styles from '@patternfly/react-styles/css/components/Menu/menu';
import { css } from '@patternfly/react-styles';
...
            label={
              <h1 className={css(styles.menuGroupTitle)}>
                <KeyIcon /> Populate the form with credentials from your selected data connection
              </h1>
            }
```